### ISSUE TITLE
Mlflow tracking server error when exploring artifacts

### ISSUE THREAD

**Issue by TsvetanRangelov** [mlflow/mlflow#3079]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for information on what types of issues we address.

**Please fill in this installation issue template to ensure a timely and thorough response.**

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Mac OS 10.14
- **MLflow installed from (source or binary)**:
source installation
- **MLflow version (run ``mlflow --version``)**:
mlflow 1.9.1
- **Python version**:
python 3.8.3
- **Exact command to reproduce**:

### Describe the problem
Using an ftp server for artifact store and making a successful run and logging it with log_model, the model gets correctly stored on the ftp server. Exploring it via the interface however crashes when clicking the folder for the model. Going inside the container the model is correctly stored. 

### Other info / logs
mlflow server -p 8000 -h 0.0.0.0 --backend-store-uri postgresql://xx:xx@postgres/mlflow_db --default-artifact-root ftp://xx:xx@ftpserver/home/mlflow
is the command for the mlflow tracking server docker container called training

`
mlflow.tracking.set_tracking_uri("http://training:8000")
`
`
with mlflow.start_run(run_name="experiment_name"):
`
`
    mlflow.log_metric("1", 1)
`
`
    mlflow.pyfunc.log_model("", python_model=a)
`
`

**Comment by dmatrix** [comment]
@TsvetanRangelov Thanks for filing this.

Can you elaborate what do you mean the interface crashes? Does the `mlflow ui` interface crashes? Do you any stacktrace? From what  read, it seems like the `mlflow ui` comes up okay and you can peruse and view the metric, but "crashes" when clicking on the model artifact.

Can you saving with `mlflow.pyfunc.log_model("pyfunc_model", python_model=a)`?

**Comment by TsvetanRangelov** [comment]
@dmatrix 
It shows the page with something went wrong and a splash picture. I cannot find any relevant information to the error on the client side and there is no relevant log on the server side relating to an exception. The js response to the click contains these files which I can verify exist.
files: [{path: "/home/mlflow/0/8b4569d7a578423b9ced8b5750c3a317/artifacts/model/MLmodel", is_dir: false,…},…]
0: {path: "/home/mlflow/0/8b4569d7a578423b9ced8b5750c3a317/artifacts/model/MLmodel", is_dir: false,…}
1: {path: "/home/mlflow/0/8b4569d7a578423b9ced8b5750c3a317/artifacts/model/conda.yaml", is_dir: false,…}
2: {path: "/home/mlflow/0/8b4569d7a578423b9ced8b5750c3a317/artifacts/model/python_model.pkl",…}

I renamed the model and it still crashes, although the paths don't look like they are different, which I am not sure why...
files: [{path: "/home/mlflow/0/d8a1707f4fa241cb90a6a6f5789e3082/artifacts/model/MLmodel", is_dir: false,…},…]
0: {path: "/home/mlflow/0/d8a1707f4fa241cb90a6a6f5789e3082/artifacts/model/MLmodel", is_dir: false,…}
1: {path: "/home/mlflow/0/d8a1707f4fa241cb90a6a6f5789e3082/artifacts/model/conda.yaml", is_dir: false,…}
2: {path: "/home/mlflow/0/d8a1707f4fa241cb90a6a6f5789e3082/artifacts/model/python_model.pkl",…}

**Comment by harupy** [comment]
@TsvetanRangelov Can you open the dev console on the browser and check if there are error messages?

**Comment by TsvetanRangelov** [comment]
Oh, there actually is an error! Here is the stack trace
```
Error: Can't find child.
    at main.dd34ca00.chunk.js:1
    at Array.forEach (<anonymous>)
    at Function.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at d (2.3c607361.chunk.js:1)
    at beginWork (2.3c607361.chunk.js:1)
    at i (2.3c607361.chunk.js:1)
n @ react_devtools_backend.js:562
_n @ 2.3c607361.chunk.js:1
commitErrorLogging @ 2.3c607361.chunk.js:1
C @ 2.3c607361.chunk.js:1
w @ 2.3c607361.chunk.js:1
_ @ 2.3c607361.chunk.js:1
b @ 2.3c607361.chunk.js:1
interactiveUpdates @ 2.3c607361.chunk.js:1
xr @ 2.3c607361.chunk.js:1
react_devtools_backend.js:562 Error: Can't find child.
    at main.dd34ca00.chunk.js:1
    at Array.forEach (<anonymous>)
    at Function.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at t.value (main.dd34ca00.chunk.js:1)
    at d (2.3c607361.chunk.js:1)
    at beginWork (2.3c607361.chunk.js:1)
    at i (2.3c607361.chunk.js:1)
n @ react_devtools_backend.js:562
value @ main.dd34ca00.chunk.js:1
commitErrorLogging @ 2.3c607361.chunk.js:1
C @ 2.3c607361.chunk.js:1
w @ 2.3c607361.chunk.js:1
_ @ 2.3c607361.chunk.js:1
b @ 2.3c607361.chunk.js:1
interactiveUpdates @ 2.3c607361.chunk.js:1
xr @ 2.3c607361.chunk.js:1
main.dd34ca00.chunk.js:1 Uncaught (in promise) TypeError: Cannot read property 'children' of undefined
    at main.dd34ca00.chunk.js:1
    at Array.forEach (<anonymous>)
    at artifactsByRunUuid (main.dd34ca00.chunk.js:1)
    at 2.3c607361.chunk.js:1
    at 2.3c607361.chunk.js:1
    at m (2.3c607361.chunk.js:1)
    at 2.3c607361.chunk.js:1
    at 2.3c607361.chunk.js:1
    at dispatch (2.3c607361.chunk.js:1)
    at 2.3c607361.chunk.js:1
```

**Comment by harupy** [comment]
It looks like this line causes the error:

https://github.com/mlflow/mlflow/blob/add51b931951dfc77716da3bd04c87c3684b4f97/mlflow/server/js/src/experiment-tracking/reducers/Reducers.js#L292

**Comment by dmatrix** [comment]
@TsvetanRangelov 
How are you launching the `mflow ui`?  Are you using`mlflow ui --backend-store-uri postgresql://xx:xx@postgres/mlflow_db`?

**Comment by TsvetanRangelov** [comment]
I am launching a tracking server via:
`
mlflow server -p 8000 -h 0.0.0.0 --backend-store-uri postgresql://xx:xx@postgres/mlflow_db --default-artifact-root ftp://xx:xx@ftpserver/home/mlflow
`
and just opening it on localhost

**Comment by gitE0Z9** [comment]
same issue here, is there any update?

**Comment by harupy** [comment]
@TsvetanRangelov Can you share your docker file?

**Comment by TsvetanRangelov** [comment]
```
version: '3'
services:
  ftpserver:
    image: fauria/vsftpd:latest
    volumes:
      - /path/to/volume/:/home/mlflow
    expose:
      - "21000-21010"
    environment:
      PASV_ADDRESS_ENABLE: "YES"
      PASV_ADDRESS: ftpserver
      FTP_USER: mlflow
      PASV_ADDR_RESOLVE: "YES"
      FTP_PASS: 123
    restart: on-failure
  postgres:
    image: postgres
    restart: always
    environment:
      POSTGRES_USER: mlflow_user
      POSTGRES_PASSWORD: example
      POSTGRES_DB: mlflow_db
  training:
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      - postgres
      - ftpserver
    command: mlflow server -p 8000 -h 0.0.0.0 --backend-store-uri postgresql://mlflow_user:example@postgres/mlflow_db --default-artifact-root ftp://mlflow:123@ftpserver/home/mlflow
    expose:
     - "8000"
    ports:
      - "8000:8000"
  test:
    depends_on:
      - postgres
      - training
      - ftpserver
    build:
      context: .
      dockerfile: Dockerfile
    command: bash -c "python example_model.py"
```


in example_model.py:
```
import mlflow
import mlflow.pyfunc

class ExampleModel(mlflow.pyfunc.PythonModel):
    def __init__(self):
        pass
    def predict(self, context, data):
        return 1

mlflow.tracking.set_tracking_uri("http://localhost:8000")
loc = mlflow.get_artifact_uri()


mlflow.end_run()

example_model = ExampleModel()

with mlflow.start_run(run_name="example_name"):
    mlflow.log_metric("1", 1)
    mlflow.pyfunc.log_model("model", python_model=example_model)


```


Dockerfiles just set up libraries nothing much

**Comment by harupy** [comment]
@TsvetanRangelov Thanks for the docker-compose.yml and python script. The root cause is [`ftp.nlst`](https://github.com/mlflow/mlflow/blob/065550f36372f8f8ea2b87b9d251d0e63c2b26a7/mlflow/store/artifact/ftp_artifact_repo.py#L113) returns abosolute paths instead of relative paths for vsftpd like below (you can firm this in the devconsole on the browser).

![Screen Shot 2020-07-30 at 20 57 28](https://user-images.githubusercontent.com/17039389/88920206-65f2be00-d2a7-11ea-858d-0950d53d2234.png)

**Comment by harupy** [comment]
Using [`stilliard/pure-ftpd`](https://github.com/stilliard/docker-pure-ftpd) image instead of `fauria/vsftpd` sovles the issue for me.

**Comment by harupy** [comment]
Fixed by #3210
